StaticAnalyzer/views/ios/common_rules.py

```diff
@@ -0,0 +1,137 @@
+"""
+Rule Format.
+
+1. desc - Description of the findings
+
+2. type
+   a. string
+   b. regex
+
+3. match
+   a. single_regex - if re.findall(regex1, input)
+   b .regex_and - if re.findall(regex1, input) and re.findall(regex2, input)
+   c. regex_or - if re.findall(regex1, input) or re.findall(regex2, input)
+   d. single_string - if string1 in input
+   e. string_and - if (string1 in input) and (string2 in input)
+   f. string_or - if (string1 in input) or (string2 in input)
+   g. string_and_or -  if (string1 in input) and ((string_or1 in input)
+                       or (string_or2 in input))
+   h. string_or_and - if (string1 in input) or ((string_and1 in input)
+                      and (string_and2 in input))
+
+4. level
+   a. high
+   b. warning
+   c. info
+   d. good
+
+5. input_case
+   a. upper
+   b. lower
+   c. exact
+
+6. others
+   a. string<no> - string1, string2, string3, string_or1, string_and1
+   b. regex<no> - regex1, regex2, regex3
+
+"""
+COMMON_RULES = [
+    {
+        'desc': 'IP Address disclosure',
+        'type': 'regex',
+        'regex1': r'\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}',
+        'level': 'warning',
+        'match': 'single_regex',
+        'input_case': 'exact',
+        'cvss': 4.3,
+        'cwe': 'CWE-200',
+        'owasp': '',
+    },
+    {
+        'desc': ('Files may contain hardcoded sensitive'
+                 ' informations like usernames, passwords, keys etc.'),
+        'type': 'regex',
+        'regex1': (r'(password\s*=\s*[\'|\"].+[\'|\"]\s{0,5})|'
+                   r'(pass\s*=\s*[\'|\"].+[\'|\"]\s{0,5})|'
+                   r'(username\s*=\s*[\'|\"].+[\'|\"]\s{0,5})|'
+                   r'(secret\s*=\s*[\'|\"].+[\'|\"]\s{0,5})|'
+                   r'(key\s*=\s*[\'|\"].+[\'|\"]\s{0,5})'),
+        'level': 'high',
+        'match': 'single_regex',
+        'input_case': 'lower',
+        'cvss': 7.4,
+        'cwe': 'CWE-312',
+        'owasp': 'M9: Reverse Engineering',
+    },
+    {
+        'desc': ('App uses SQLite Database. '
+                 'Sensitive Information should be encrypted.'),
+        'type': 'string',
+        'string1': 'sqlite3_exec',
+        'level': 'info',
+        'match': 'single_string',
+        'input_case': 'exact',
+        'cvss': 0,
+        'cwe': '',
+        'owasp': '',
+    },
+    {
+        'desc': ('User input in "loadHTMLString" '
+                 'will result in JavaScript Injection.'),
+        'type': 'string',
+        'string1': 'loadHTMLString',
```

Comment:
Is this applicable to Swift?

Reply:
Yes.
Reference: https://github.com/OWASP/owasp-mstg/blob/master/Document/0x06h-Testing-Platform-Interaction.md#testing-how-webviews-are-loaded